compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3298,7 +3299,9 @@ public void visit(BLangQueryExpr queryExpr) {
             } else if (kind == NodeKind.ON_CONFLICT) {
                 typeCheckOnConflictClause((BLangOnConflictClause) clause, parentEnv);
             } else if (kind == NodeKind.WHERE) {
-                whereEnv = typeCheckWhereClause((BLangWhereClause) clause, select, parentEnv);
+                whereEnv = typeCheckWhereClause(((BLangWhereClause) clause).expression, select, parentEnv);
+            }else if (kind == NodeKind.ON) {
```

Comment:
Please format the added change

Reply:
Formatting issues have been rectified in https://github.com/ballerina-platform/ballerina-lang/pull/23252